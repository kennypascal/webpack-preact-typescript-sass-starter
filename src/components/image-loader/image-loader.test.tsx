import { h } from 'preact';
import { shallow } from 'enzyme';
import ImageLoader, { DURATION } from './image-loader';

const mockSrc = 'https://picsum.photos/seed/picsum/200/300'
const mockAlt = 'A happy test.'

let wrapper;

beforeEach(() => {
  jest.useFakeTimers();
  wrapper = shallow(<ImageLoader src={mockSrc} alt={mockAlt} />);
});


describe('ImageLoader', () => {
  it('should correctly render image-loader component when loading', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should correctly render image-loader component when loaded', () => {
    wrapper.find('img').props().onLoad();
    wrapper.update();
    expect(wrapper.state().status).toBe('loaded');
    expect(wrapper.state().isReady).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it('should correctly render image-loader component when loaded and ready', () => {
    wrapper.find('img').props().onLoad();
    wrapper.update();
    jest.advanceTimersByTime(DURATION * 2);
    expect(wrapper.state().isReady).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it('should correctly render image-loader component when error', () => {
    wrapper.find('img').props().onError();
    wrapper.update();
    expect(wrapper.state().status).toBe('error');
    expect(wrapper.state().isReady).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it('should correctly render image-loader component when error and ready', () => {
    wrapper.find('img').props().onError();
    wrapper.update();
    jest.advanceTimersByTime(DURATION * 2);
    expect(wrapper.state().isReady).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});