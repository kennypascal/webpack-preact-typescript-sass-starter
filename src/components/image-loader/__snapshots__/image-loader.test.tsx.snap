// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLoader should correctly render image-loader component when error 1`] = `
<div
  className="image-loader"
>
  <p>
    error
  </p>
  <img
    alt="A happy test."
    className="image image--error image--in"
    onError={[Function]}
    onLoad={[Function]}
    src="https://picsum.photos/seed/picsum/200/300"
  />
</div>
`;

exports[`ImageLoader should correctly render image-loader component when error and ready 1`] = `
<div
  className="image-loader"
>
  <p>
    error
  </p>
  <img
    alt="A happy test."
    className="image image--error image--in"
    onError={[Function]}
    onLoad={[Function]}
    src="https://picsum.photos/seed/picsum/200/300"
  />
</div>
`;

exports[`ImageLoader should correctly render image-loader component when loaded 1`] = `
<div
  className="image-loader"
>
  <p>
    loaded
  </p>
  <img
    alt="A happy test."
    className="image image--loaded image--in"
    onError={[Function]}
    onLoad={[Function]}
    src="https://picsum.photos/seed/picsum/200/300"
  />
</div>
`;

exports[`ImageLoader should correctly render image-loader component when loaded and ready 1`] = `
<div
  className="image-loader"
>
  <img
    alt="A happy test."
    className="image image--loaded image--in"
    onError={[Function]}
    onLoad={[Function]}
    src="https://picsum.photos/seed/picsum/200/300"
  />
</div>
`;

exports[`ImageLoader should correctly render image-loader component when loading 1`] = `
<div
  className="image-loader"
>
  <p>
    loading
  </p>
  <img
    alt="A happy test."
    className="image image--loading image--in"
    onError={[Function]}
    onLoad={[Function]}
    src="https://picsum.photos/seed/picsum/200/300"
  />
</div>
`;
